src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php

```diff
@@ -118,4 +223,211 @@ public function iShouldBeNotifiedThatItHasBeenSuccessfullyCreated(): void
     {
         Assert::true($this->responseChecker->isCreationSuccessful($this->client->getLastResponse()));
     }
+
+    /**
+     * @Then I should be notified that it has been successfully edited
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyEdited(): void
+    {
+        Assert::true(
+            $this->responseChecker->isUpdateSuccessful($this->client->getLastResponse()),
+            'Product could not be edited'
+        );
+    }
+
+    /**
+     * @Then I should be notified that this product is in use and cannot be deleted
+     */
+    public function iShouldBeNotifiedThatThisProductIsInUseAndCannotBeDeleted(): void
+    {
+        Assert::false(
+            $this->responseChecker->isDeletionSuccessful($this->client->getLastResponse()),
+            'Product can be deleted, but it should not'
+        );
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully deleted
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyDeleted(): void
+    {
+        Assert::true(
+            $this->responseChecker->isDeletionSuccessful($this->client->getLastResponse()),
+            'Product still exists, but it should not'
+        );
+    }
+
+    /**
+     * @Then I should see :numberOfProducts products in the list
+     */
+    public function iShouldSeeProductsInTheList(int $count): void
+    {
+        Assert::count($this->responseChecker->getCollection($this->client->getLastResponse()), $count);
+    }
+
+    /**
+     * @Then I should( still) see a product with :field :value
+     */
+    public function iShouldSeeProductWith(string $field, string $value): void
+    {
+        Assert::true(
+            $this
+                ->responseChecker
+                ->hasItemWithTranslation($this->client->getLastResponse(), 'en_US', $field, $value)
+            ,
+            sprintf('Product has not %s with %s', $field, $value)
+        );
+    }
+
+    /**
+     * @Then I should not see any product with :field :value
+     */
+    public function iShouldNotSeeAnyProductWith(string $field, string $value): void
+    {
+        $this->client->update();
+
+        Assert::false(
+            $this
+                ->responseChecker
+                ->hasItemWithTranslation($this->client->getLastResponse(), 'en_US', $field, $value)
+            ,
+            sprintf('Product with %s set as %s still exists, but it should not', $field, $value)
+        );
+    }
+
+    /**
+     * @Then I should not be able to edit its code
+     */
+    public function iShouldNotBeAbleToEditItsCode(): void
+    {
+        $this->client->addRequestData('code', '_NEW');
+        $response = $this->client->index();
```

Comment:
There is no need for this variable

Reply:
Can't we check the current code on the show page?

Reply:
I remove index() it was my fault, but i add update because in this step I need to check all flow for check that is possible changing code

Scenario:
![Screenshot 2020-03-18 at 11 08 45](https://user-images.githubusercontent.com/39232096/76951828-91829680-690c-11ea-900f-e0b23d688929.png)

Similar situation in ExchangeRate : https://github.com/Sylius/Sylius/blob/master/src/Sylius/Behat/Context/Api/Admin/ManagingExchangeRatesContext.php#L346